userland/AppMakefile.mk

```diff
@@ -66,19 +100,19 @@ $$(BUILDDIR)/$(1):
 
 # First step doesn't actually compile, just generate header dependency information
 # More info on our approach here: http://stackoverflow.com/questions/97338
-$$(BUILDDIR)/$(1)/%.o: %.c | $$(BUILDDIR)/$(1)
+$$(BUILDDIR)/$(1)/%.o: %.c | $$(BUILDDIR)/$(1) $$(HDRS)
```

Comment:
Their directories are being `-I` included by the same thing that added them to the `HDRS` variable in the first place.

Now, a more compelling argument is with those same rules, you could add:

     $(SRCS_WITH_HDR_DEPS): $(HDRS)

as a rule, instead of here, which is marking an artificially high number of source files as dependent on those headers. I can make that change